irohad/main/impl/on_demand_ordering_init.cpp

```diff
@@ -117,12 +117,18 @@ namespace iroha {
 
         matchEvent(latest_commit, on_blocks, on_nothing);
 
-        auto peer = [this](auto round, auto pos) {
-          auto &permutation = permutations_[round];
+        const auto current_block_round = latest_commit.round.block_round + 1;
+        auto peer = [this, &current_block_round](auto block_round_advance,
```

Comment:
I suggest renaming the handler because it clashes with name `peer` which introduces at 126 line.

Reply:
ok, although it is not the purpose of the PR.